builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -354,10 +354,12 @@ func resourceArmVirtualMachine() *schema.Resource {
 						"provision_vm_agent": {
 							Type:     schema.TypeBool,
 							Optional: true,
+							Default:  false,
```

Comment:
Checking [the Azure docs](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-create-or-update) & #9995 - the default value for the `provisionVMAgent` field should be true if it's unset:

> When this property is not specified in the request body, default behavior is to set it to true. This will ensure that VM Agent is installed on the VM so that extensions can be added to the VM later.

I'll update this and look into a schema migration